### PR TITLE
Messages and bug fixes

### PR DESCRIPTION
**Pull request by based-kwl**
Changelog:
- Socket implementation
- Messages POST routes
  - `/send`
- Messages GET routes
  - `/:mID`
  - `/bulk/:sID/:amount`
  - `/bulk/:sID/:ignore`
- Automatic culling of messages embedded within a studyroom. By default a studyroom will contain at most 30 messages. Retrieve more using the bulk message retrieval routes.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,8 +27,11 @@
     "express": "^4.18.1",
     "jsonwebtoken": "^8.5.1",
     "mongodb-memory-server": "^8.9.5",
-    "mongoose": "^6.7.1",
-    "nodemon": "^2.0.20"
+    "mongoose": "^6.6.5",
```
Comment (guaj) [inline package.json:30]: why downgrade the version ?

File: src/routes/room.js [hunk 2]
```diff
@@ -93,24 +90,65 @@ router.route('/add').post(async(req, res)=>{
...
     { sID: ID }, 
     { participants: participants },
   );
   const student = await Student.findOne( { email: email })
-  const StudyRooms=student.StudyRooms.push(sID)
+  var studyRooms=student.StudyRooms;
+  studyRooms.push(ID);
 
-  Student.updateOne(
+  await Student.updateOne(
```
Comment (guaj) [inline src/routes/room.js:110]: the await here has no effects

**Review by guaj — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (based-kwl) [reply package.json:30]: No idea I didn't touch that. Probably because this branch is fairly old

File: src/routes/room.js (hunk 2 above)
Comment (based-kwl) [reply src/routes/room.js:110]: removing the await here causes student to be undefined 
image

**Pull request merged**